test/rest_api/test_storage_service.py

```diff
@@ -453,11 +453,17 @@ def test_range_to_endpoint_map_tablets_disabled_keyspace_param_only(cql, this_dc
         resp = rest_api.send("GET", f"storage_service/range_to_endpoint_map/{keyspace}")
         resp.raise_for_status()
 
-@pytest.mark.xfail(reason="rest_api suite doesn't support tablets yet (#17338), run test manually")
-def test_describe_ring(cql, this_dc, rest_api):
+def test_describe_ring(cql, this_dc, rest_api, has_tablets):
```

Comment:
In the future, please consider not adding an xfail in one patch and removing it in this patch. It's confusing to see a dozen xfail in one patch not knowing why - and later realizing that most of them are removed.

Reply:
I assumed that marking failing tests and then fixing them in turn would make it easier to read, but I will keep in mind

Reply:
Maybe other people think differently, but for me, it's just double the effort: First I need to understand why you change a dozen specific tests (and not others) to xfail, and then I see how you fix them. Could have just fixed them and leave as the last patch a few more unfixed tests as xfail if needed.

But it's not important now. I don't want you to waste your time to restructure your PR just to make it conform to my prefered style.